packages/ReactDagEditor/stories/components/FeaturesDemo.tsx

```diff
@@ -245,14 +243,17 @@ class MyPortConfig implements IPortConfig {
   }
 }
 
+const graphConfig = GraphConfigBuilder.default()
```

Comment:
It will make it difficult to modify the config at runtime.

How about

```typescript
<NodeShapeProvider renderNode={} {/*more props*/} >{children}</NodeShapeProvider >
```

Reply:
In this approach it is much more easier for scoped shapes.

Reply:
The problem is that reducers require the configs to do some calculation, thus it's unsafe if we put configs in components or contexts.
So I moved it to the state
The code by me design would be like the following after lifting state is finished:
```
const nextConfig = GraphConfigBuilder.from(prevConfig).registerPort("xxx", xxx).build()
dispatch({
  type: UpdateSetttings,
  graphConfig: nextConfig 
});
```